src/adaptors/adaptor_common.c

```diff
@@ -23,150 +23,68 @@
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 8192;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 16384;
+size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 32768;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 65536;
 
 
-        if (!!config_ssl_profile->ssl_ciphers) {
-            res = pn_tls_config_set_impl_ciphers(*tls_config, config_ssl_profile->ssl_ciphers);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to set tls ciphers (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-
-
-        if (is_listener) {
-            if (config->authenticate_peer) {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-            else {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_ANONYMOUS_PEER, 0);
-            }
-        }
-        else {
-            // Connector.
-            if (config->verify_host_name) {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER_NAME, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-            else {
-                res = pn_tls_config_set_peer_authentication(*tls_config, PN_TLS_VERIFY_PEER, config_ssl_profile->ssl_trusted_certificate_db);
-            }
-
-            *tls_has_output = true; // always true for initial client side TLS.
-        }
+qd_adaptor_buffer_t *qd_adaptor_buffer()
+{
+    qd_adaptor_buffer_t *adaptor_buff = new_qd_adaptor_buffer_t();
+    ZERO(adaptor_buff);
+    DEQ_ITEM_INIT(adaptor_buff);
+    return adaptor_buff;
+}
 
-        if (res != 0) {
-            qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] Unable to set tls peer authentication for sslProfile %s - (%d)", conn_id, config->ssl_profile_name, res);
-            break;
-        }
+qd_adaptor_buffer_t *qd_adaptor_buffer_raw(pn_raw_buffer_t *buffer)
+{
+    qd_adaptor_buffer_t *adaptor_buff = qd_adaptor_buffer();
+    ZERO(buffer);
+    buffer->bytes = (char*) qd_adaptor_buffer_base(adaptor_buff);
+    buffer->capacity = qd_adaptor_buffer_capacity(adaptor_buff);
+    buffer->size = 0;
+    buffer->offset = 0;
+    buffer->context = (uintptr_t) adaptor_buff;
+    return adaptor_buff;
+}
 
-        //
-        // Provide an ordered list of application protocols for ALPN by calling pn_tls_config_set_alpn_protocols. In our case, h2 is the only supported protocol.
-        // A list of protocols can be found here - https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.txt
-        //
-        if (protocols)
-            pn_tls_config_set_alpn_protocols(*tls_config, protocols, NUM_ALPN_PROTOCOLS);
+qd_adaptor_buffer_t *qd_adaptor_buffer_list_append(qd_adaptor_buffer_list_t *buflist, const uint8_t *data, size_t len)
+{
+    //
+    // If len is zero, there's no work to do.
+    //
+    if (len == 0)
+        return DEQ_TAIL(*buflist);
```

Comment:
Why return just the tail buffer?  This signature isn't very intuitive - the corresponding qd_buffer_list_append() api returns nothing (void).
If the caller needs the tail buffer it should use DEQ_TAIL on the list.